src/sql/src/catalog.rs

```diff
@@ -248,12 +248,8 @@ pub enum CatalogError {
     UnknownSchema(String),
     /// Unknown item.
     UnknownItem(String),
-    /// Invalid attempt to depend on a sink.
-    InvalidSinkDependency(String),
-    /// Invalid attempt to depend on an index.
-    InvalidIndexDependency(String),
-    /// Invalid attempt to depend on a type.
-    InvalidTypeDependency(String),
+    /// Invalid attempt to depend on a non-dependable item.
+    InvalidDependency(String, &'static str),
```

Comment:
I think you might as well use `CatalogItemType` as the second thing here?

Btw, I try to avoid tuple variants with more than one element pretty aggressively, since it's hard to tell at a glance what the fields are.